AK/Random.cpp

```diff
@@ -1,98 +1,126 @@
+        // EINVAL, invalid flags provided, which there aren't.
+        // ENOSYS, the kernel doesn't implement the syscall. The syscall has been available since Linux 3.17, which was released in 2019.
+        // EFAULT, the address is outside the accessible space; the caller is responsible for providing a valid address.
+        //
+        // Therefore, the only somewhat plausible errors are EFAULT and EINTR.
+        // Nonetheless, we will check for the possibility of all the error codes out of an abundance of caution.
+        ret = getrandom(buf, size, 0);
 
-#    if defined(__unix__)
-    // The maximum permitted value for the getentropy length argument.
-    static constexpr size_t getentropy_length_limit = 256;
-    auto iterations = bytes.size() / getentropy_length_limit;
+        if (ret == -1 && errno != EINTR) [[unlikely]]
+            return Error::from_errno(errno);
 
-    for (size_t i = 0; i < iterations; ++i) {
-        if (getentropy(bytes.data(), getentropy_length_limit) != 0) {
-            fill_with_random_fallback();
-            return;
+        // Per the manual:
+        //
+        // > On success, getrandom() returns the number of bytes that were
+        // > copied to the buffer buf.  This may be less than the number of
+        // > bytes requested via buflen if either GRND_RANDOM was specified in
+        // > flags and insufficient entropy was present in the random source or
+        // > the system call was interrupted by a signal.
+        //
+        // > The user of getrandom() *must* always check the return value, to
+        // > determine whether either an error occurred or fewer bytes than
+        // > requested were returned.
+        if (ret > 0) [[likely]] {
+            size -= ret;
+            buf += ret;
         }
-
-        bytes = bytes.slice(getentropy_length_limit);
     }
+#elif defined(AK_OS_WINDOWS)
+    if (size > NumericLimits<u32>::max()) [[unlikely]]
+        return Error::from_string_literal("The input is too large");
 
-    if (bytes.is_empty() || getentropy(bytes.data(), bytes.size()) == 0)
-        return;
-#    elif defined(AK_OS_WINDOWS)
-
-    if (bytes.size() > NumericLimits<u32>::max()) [[unlikely]] {
-        fill_with_random_fallback();
-        return;
+    unsigned char* ptr = reinterpret_cast<unsigned char*>(const_cast<void*>(buf));
+    NTSTATUS status = ::BCryptGenRandom(BCRYPT_RNG_ALG_HANDLE, ptr, size, 0);
+    if (!NT_SUCCESS(status)) [[unlikely]] {
+        ULONG error = RtlNtStatusToDosError(status);
+        return Error::from_windows_error(error);
     }
+#else
+    // There shouldn't be a build target where this can happen.
+    //
+    // As for MacOS and iOS, they appear to both support arc4random_buf, and AK/Platform.h defines AK_OS_BSD_GENERIC when building for them,
+    // so they will be covered by the compile target that uses arc4random_buf.
```

Comment:
This note feels a bit out of place here. If you're worried that people might misunderstand the first `#if defined(..)`, add a comment there to explain it includes macOS and iOS.